### PR TITLE
Rely on the NATS cluster connection status when reporting the Eventing CR status

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Rely on the NATS cluster connection status when reporting the Eventing CR status.
As part of this PR, remove unsupported conditions from the Eventing Status. Because currently we preserve conditions even though it is removed in old releases.

**Related issue(s)**

- https://github.com/kyma-project/eventing-manager/issues/267.

### CODE REVIEW

**Comment by marcobebway** [comment]
Remaining tasks:

- [ ] Fix existing tests.
- [ ] Add more tests if needed.

File: internal/connection/nats/interface.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//go:generate go run github.com/vektra/mockery/v2 --name=Interface --structname=Connection --filename=connection.go
+type Interface interface {
+	// Connect connects to NATS and returns an error if it cannot connect.
+	Connect() error
+
+	// Disconnect disconnects the NATS connection.
+	Disconnect()
+
+	// RegisterReconnectHandlerIfNotRegistered registers a ReconnectHandler only if it was not registered before.
+	RegisterReconnectHandlerIfNotRegistered(natsio.ConnHandler)
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply internal/connection/nats/interface.go]: Done

File: internal/connection/nats/interface.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	// RegisterDisconnectErrHandlerIfNotRegistered registers a DisconnectErrHandler only if it was not registered before.
+	RegisterDisconnectErrHandlerIfNotRegistered(natsio.ConnErrHandler)
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply internal/connection/nats/interface.go]: Done

File: internal/controller/operator/eventing/controller.go [hunk 3]
```diff
@@ -492,99 +502,139 @@ func (r *Reconciler) handleEventingReconcile(ctx context.Context,
...
 	}
 
-	// check nats CR if it exists and is in natsAvailable state
-	err = r.checkNATSAvailability(ctx, eventing)
-	if err != nil {
-		return kctrl.Result{}, r.syncStatusWithNATSErr(ctx, eventing, err, log)
+	if connErr := r.connectToNATS(eventingCR); connErr != nil {
+		if errors.Is(connErr, ErrNilNATSConnectionBuilder) {
+			r.namedLogger().Error(connErr)
+			os.Exit(1)
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply internal/controller/operator/eventing/controller.go]: Removed

File: internal/controller/operator/eventing/controller.go [hunk 4]
```diff
@@ -492,99 +502,139 @@ func (r *Reconciler) handleEventingReconcile(ctx context.Context,
...
+		}
+
+		if errors.Is(connErr, natsconnectionerrors.ErrCannotConnect) {
+			return kctrl.Result{}, reconcile.TerminalError(
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply internal/controller/operator/eventing/controller.go:577]: Yes we stop reconciling the Eventing CR, however the nats client will keep trying to connect.

**Review by muralov — APPROVED** [review]

**Pull request merged**